.github/workflows/windows.yml

```diff
@@ -4,28 +4,32 @@ on: [pull_request, push]
 
 jobs:
   shared:
-    runs-on: windows-latest
     # Run a job for each of the specified target architectures:
     strategy:
       matrix:
-        arch:
-          - win64
-          - win32
+        os:
+          - windows-latest
+          - windows-2016
+        platform:
+          - arch: win64
+            config: VC-WIN64A enable-fips
+          - arch: win32
+            config: VC-WIN32 --strict-warnings no-fips
```

Comment:
This may save some time doing no-fips - but I wonder if it is a good idea

Reply:
Yeah...  but that's what was already in place in the previous PS `if` statement, so I just followed what was already there.
It's worth reassessing, but not sure if that should happen in this PR

Reply:
I assume, BTW, that this was made no-fips because VC-WIN32 builds take a huge amount of time already...  VC-WIN32 *without* FIPS takes more time than VC-WIN64A *with* FIPS...

Reply:
I do not think we should change this. Instead we could have a separate more complete workflow for 32 bit builds on pushes to master only.